hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/BaseHoodieWriteClient.java

```diff
@@ -1142,7 +1152,13 @@ protected boolean scheduleCleaningAtInstant(String instantTime, Option<Map<Strin
 
   private Option<String> scheduleTableServiceInternal(String instantTime, Option<Map<String, String>> extraMetadata,
                                                       TableServiceType tableServiceType) {
+    if (!tableServicesEnabled(config)) {
+      return Option.empty();
+    }
     switch (tableServiceType) {
+      case ARCHIVE:
+        LOG.info("Scheduling archiving is not supported. Skipping.");
```

Comment:
why adding this? no one should be scheduling archiving. so, wondering if we should not add this and let it throw in default clause w/ unknown table service type ?

Reply:
@nsivabalan it's because `org.apache.hudi.common.model.TableServiceType` was updated to include ARCHIVE, so the switch should enumerate the list for completeness.